File: LeetCode/1854. Maximum Population Year/solution.py
class Solution:
    def maximumPopulation(self, logs: List[List[int]]) -> int:
        delta = defaultdict(int)
        for a, b in logs:
            delta[a] += 1
            delta[b] -= 1
        cur, res, year = 0, 0, -1
        for x in range(1950,2051,1):
            # print(x)
            cur += delta[x]
            if cur > res:
                res = cur
                year = x
        return year

File: LeetCode/1854. Maximum Population Year/solution.cpp
class Solution {
public:
    int maximumPopulation(vector<vector<int>>& logs) {
        map<int,int> delta;
        for (auto &log : logs){
            auto a = log[0];
            auto b = log[1];
            delta[a]++;
            delta[b]--;
        }
        int res = 0, year = -1, cur = 0;
        for (int i = 1950; i <= 2050; i++){
            cur += delta[i];
            if (cur > res){
                res = cur;
                year = i;
            }
        }
        return year;

    }
};